apps/server/src/stores/cachingStore.ts

```diff
@@ -0,0 +1,52 @@
+interface CacheData {
+  data: unknown;
+}
+
+const runtimeCache: Map<string, CacheData> = new Map();
+
+export function checkCached(key: string): boolean {
+  return Object.hasOwn(runtimeCache, key);
+}
+
+export function getCached<T>(key: string, callback: () => T): T {
+  if (!Object.hasOwn(runtimeCache, key)) {
+    try {
+      const data = callback();
+      runtimeCache.set(key, { data });
+    } catch (error) {
+      console.log(`Failed retrieving data from callback: ${error}`);
+    }
+  }
```

Comment:
Getting a missing key should probably just throw an error. Any reason why the default behavior should be setting it?

Reply:
I guess that works. When I think of it as a cache as opposed to a glorified `Map`, it makes more sense. Resolving